Added a separate 'ext4' based partion for /var/lib/docker.

agcooke opened the pull request:
*Issue #, if available:*
https://github.com/awslabs/amazon-eks-ami/issues/79

*Description of changes:*
We have large images in our monolithic application. This commit attaches a separate EBS volume for docker artefacts to accomdate them.

It is possible to specify that the image gets formatted as XFS or ext4. We have been running with ext4 for some weeks now and have not had any issues. We did have issues with XFS, but did not pinpoint as the root cause because there where many other things that we fixed.

The work is mostly copied from @albertrdixon work in the [LumosLab fork](https://github.com/lumoslabs/amazon-eks-ami)
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

agcooke commented:
This is not a backwards compatible, if we want to go ahead with it I can update the CloudFormation templates.

We had to update our terraform files to add the extra volume.

kinghajj commented:
Is the purpose of this change primarily for more storage? If so, couldn't one simply increase the root volume size? You wrote in the comments for #79, "this would take strain off the OS network IO, since all EBS volumes are actually connected over the network." But since the root volume is also EBS, shouldn't that isolation hold for it, too?

dawidmalina commented:
@micahhausler can you please take a look on this pull request?

dawidmalina commented:
@agcooke can you rebase your pull request?

I have also few questions:
- script allow us to configure FSTYPE but there is no way to configure it with packer variable
- script allow us to configure FSTAB_OPTIONS but there is no way to configure it with packer variable
- volume size is currently hard-coded but it would be nice to configure it with packer variable

@micahhausler do you think such change will be accepted by you? We are looking for this to be aligned with CIS benchmark
> 1.1 – Ensure a separate partition for containers has been created

agcooke commented:
Looks like this has been added to master in another commit: https://github.com/awslabs/amazon-eks-ami/commit/f8c117f90d055b46ff09b5e9a9c081c612316478

agcooke commented:
Oh, my mistake, that is something else...

agcooke commented:
@dawidmalina  I rebased to master and removed the variables. I have moved on to another project so difficult for me to test right now, can you take a look at it? Then we poke maintainers to get it reviewed...

dawidmalina commented:
@agcooke recently added ability to add custom source ami during build https://github.com/awslabs/amazon-eks-ami/pull/203 gives me no reasons to continue this change. I think that there are too many corner cases related to operation system configuration (like hardening, different partition layout, etc) so IMHO we should keep here only changes related to k8s node setup. If someone is not satisfied with default source ami - no problem it can be easily changed now.

Anyway final decision must be done by @micahhausler, @mogren
